components/image_viewer/image_viewer.vue

```diff
@@ -171,27 +186,39 @@ export default {
         }
       },
     },
+
+    open: {
+      handler: function (open) {
+        if (open !== null) {
+          this.isOpen = open;
+        }
+      },
+
+      immediate: true,
+    },
   },
 
   methods: {
-    open () {
-      this.show = true;
+    openModal () {
+      this.isOpen = true;
+      this.$emit('update:open', true);
```

Comment:
in addition to update:open you should also trigger a normal "open" event that is documented and meant to be captured by the user.